### PR TITLE
EPMDJ-2228: .NET agent uses websocket for communication

### PR DESCRIPTION
**Pull request by SunFellow**
Drill4J .NET agent now uses agent-connector DLL for communication with Admin via websocket.
Kotlin native agent connector libraries are built from sibling repository:
https://github.com/Drill4J/native-agent-connector
and are published to Artifactory under this URL:
https://oss.jfrog.org/artifactory/list/oss-release-local/com/epam/drill/dotnet
.NET agent uses Windows native DLL built by mingw for X64 platform.
It needs C Header (`agent_connector_api.h`) and DLL `agent_connector.dll` (Release and Debug).

* As a first step added `download_nac.bat` script that downloads and unpacks required artifacts. It supposes that curl and unzip are preinstalled at the environment.
* New project `Connector.vcxproj` wraps usage of `agent_connector.dll` - it loads the library and resolves exports. It exposes `IConnector` interface to allow mocking. It also builds `Connector.exe`, a console test application, that establishes connection and listens for messages from admin.
* Integrated `agent_connector.dll` to Drill4dotNet to `class ProClient` that will send / receive messages from admin and will control work of `class CProfilerCallback`
* Added `class ConnectorMock` to Unit tests to allow testing w/o the real DLL.

### CODE REVIEW

File: Drill4dotNet/Connector/Connector.cpp [hunk 1]
```diff
@@ -16,17 +16,152 @@
...
+            }
+
+            void operator()(const HMODULE library) const noexcept
+            {
+                if (library == nullptr)
+                {
+                    return;
+                }
+
+                if (::FreeLibrary(library) != 0)
```
Comment (SunFellow) [inline Drill4dotNet/Connector/Connector.cpp:40]: Error: FreeLibrary != 0 means success.
Style: Declaration is: 
```
BOOL FreeLibrary(
  HMODULE hLibModule
);
```
So, it's better to compare against `TRUE` or `FALSE` instead `0` or non-zero.

File: Drill4dotNet/Connector/Connector.cpp [hunk 2]
```diff
@@ -16,17 +16,152 @@
...
+    {
+    private:
+        class Deleter
+        {
+        public:
+            void operator()(const HANDLE event) const noexcept
+            {
+                if (event != NULL)
+                {
+                    ::CloseHandle(event);
```
Comment (SunFellow) [inline Drill4dotNet/Connector/Connector.cpp:125]: We can add checking the result of `::CloseHandle` as well...

File: Drill4dotNet/Connector/Connector.cpp (hunk 1 above)
Comment (Aleksandr-Kapustin-epam-com) [reply Drill4dotNet/Connector/Connector.cpp:40]: I'm stupid. Will fix.

File: Drill4dotNet/Connector/Connector.cpp (hunk 2 above)
Comment (Aleksandr-Kapustin-epam-com) [reply Drill4dotNet/Connector/Connector.cpp:125]: Will fix.

**Pull request merged**